sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -183,7 +183,7 @@ trait Block extends TreeNode[Block] with JavaCode {
     def doTransform(arg: Any): AnyRef = arg match {
       case e: ExprValue => transform(e)
       case Some(value) => Some(doTransform(value))
-      case seq: Traversable[_] => seq.map(doTransform)
+      case seq: Iterable[_] => seq.map(doTransform)
```

Comment:
This is the only part of the change that is of concern. It's internal, but this now only matches `Iterable` and not `Traversable`, which are not the same in Scala 2.12. However, AFAICT, no standard class we'd expect here doesn't extend `Iterable`. See https://docs.scala-lang.org/overviews/collections/overview.html ; `Seq`, `Set` and `Map` are also `Iterable` already. This shouldn't change behavior, but tests will help prove that.